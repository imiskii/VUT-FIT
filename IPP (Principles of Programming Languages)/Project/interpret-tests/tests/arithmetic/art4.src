.ippcode23

DEFVAR GF@a
DEFVAR GF@b
DEFVAR GF@c


MOVE GF@a float@10.5
MOVE GF@b float@-2.6
MOVE GF@c float@0.0



DEFVAR GF@x

ADD GF@x GF@a GF@b
WRITE GF@x
WRITE string@|

ADD GF@x float@10.8 GF@a
WRITE GF@x
WRITE string@|

ADD GF@x GF@a float@10.8 
WRITE GF@x
WRITE string@|

ADD GF@x float@50.7 float@10.3 
WRITE GF@x
WRITE string@||
###


SUB GF@x GF@a GF@b
WRITE GF@x
WRITE string@|

SUB GF@x float@-10.2 GF@a
WRITE GF@x
WRITE string@|

SUB GF@x GF@a float@10.2
WRITE GF@x
WRITE string@|

SUB GF@x float@50.6 float@10.4
WRITE GF@x
WRITE string@||

###

MUL GF@x GF@a GF@b
WRITE GF@x
WRITE string@|

MUL GF@x float@50.4 GF@b
WRITE GF@x
WRITE string@|

MUL GF@x GF@b float@50.5
WRITE GF@x
WRITE string@|

MUL GF@x float@-2.4 float@50.6
WRITE GF@x
WRITE string@||

###

IDIv GF@x GF@a GF@b
WRITE GF@x
WRITE string@|

IDIv GF@x GF@a float@10.0
WRITE GF@x
WRITE string@|

IDIv GF@x float@10.4 GF@a
WRITE GF@x
WRITE string@|

IDIv GF@x float@-5 float@-2.2
WRITE GF@x
WRITE string@||