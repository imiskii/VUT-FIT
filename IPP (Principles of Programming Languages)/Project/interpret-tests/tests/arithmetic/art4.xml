<?xml version="1.0" encoding="UTF-8"?>
<program language="IPPcode23">
 <instruction order="1" opcode="DEFVAR">
  <arg1 type="var">GF@a</arg1>
 </instruction>
 <instruction order="2" opcode="DEFVAR">
  <arg1 type="var">GF@b</arg1>
 </instruction>
 <instruction order="3" opcode="DEFVAR">
  <arg1 type="var">GF@c</arg1>
 </instruction>
 <instruction order="4" opcode="MOVE">
  <arg1 type="var">GF@a</arg1>
  <arg2 type="float">10.5</arg2>
 </instruction>
 <instruction order="5" opcode="MOVE">
  <arg1 type="var">GF@b</arg1>
  <arg2 type="float">-2.6</arg2>
 </instruction>
 <instruction order="6" opcode="MOVE">
  <arg1 type="var">GF@c</arg1>
  <arg2 type="float">0.0</arg2>
 </instruction>
 <instruction order="7" opcode="DEFVAR">
  <arg1 type="var">GF@x</arg1>
 </instruction>
 <instruction order="8" opcode="ADD">
  <arg1 type="var">GF@x</arg1>
  <arg2 type="var">GF@a</arg2>
  <arg3 type="var">GF@b</arg3>
 </instruction>
 <instruction order="9" opcode="WRITE">
  <arg1 type="var">GF@x</arg1>
 </instruction>
 <instruction order="10" opcode="WRITE">
  <arg1 type="string">|</arg1>
 </instruction>
 <instruction order="11" opcode="ADD">
  <arg1 type="var">GF@x</arg1>
  <arg2 type="float">10.8</arg2>
  <arg3 type="var">GF@a</arg3>
 </instruction>
 <instruction order="12" opcode="WRITE">
  <arg1 type="var">GF@x</arg1>
 </instruction>
 <instruction order="13" opcode="WRITE">
  <arg1 type="string">|</arg1>
 </instruction>
 <instruction order="14" opcode="ADD">
  <arg1 type="var">GF@x</arg1>
  <arg2 type="var">GF@a</arg2>
  <arg3 type="float">10.8</arg3>
 </instruction>
 <instruction order="15" opcode="WRITE">
  <arg1 type="var">GF@x</arg1>
 </instruction>
 <instruction order="16" opcode="WRITE">
  <arg1 type="string">|</arg1>
 </instruction>
 <instruction order="17" opcode="ADD">
  <arg1 type="var">GF@x</arg1>
  <arg2 type="float">50.7</arg2>
  <arg3 type="float">10.3</arg3>
 </instruction>
 <instruction order="18" opcode="WRITE">
  <arg1 type="var">GF@x</arg1>
 </instruction>
 <instruction order="19" opcode="WRITE">
  <arg1 type="string">||</arg1>
 </instruction>
 <instruction order="20" opcode="SUB">
  <arg1 type="var">GF@x</arg1>
  <arg2 type="var">GF@a</arg2>
  <arg3 type="var">GF@b</arg3>
 </instruction>
 <instruction order="21" opcode="WRITE">
  <arg1 type="var">GF@x</arg1>
 </instruction>
 <instruction order="22" opcode="WRITE">
  <arg1 type="string">|</arg1>
 </instruction>
 <instruction order="23" opcode="SUB">
  <arg1 type="var">GF@x</arg1>
  <arg2 type="float">-10.2</arg2>
  <arg3 type="var">GF@a</arg3>
 </instruction>
 <instruction order="24" opcode="WRITE">
  <arg1 type="var">GF@x</arg1>
 </instruction>
 <instruction order="25" opcode="WRITE">
  <arg1 type="string">|</arg1>
 </instruction>
 <instruction order="26" opcode="SUB">
  <arg1 type="var">GF@x</arg1>
  <arg2 type="var">GF@a</arg2>
  <arg3 type="float">10.2</arg3>
 </instruction>
 <instruction order="27" opcode="WRITE">
  <arg1 type="var">GF@x</arg1>
 </instruction>
 <instruction order="28" opcode="WRITE">
  <arg1 type="string">|</arg1>
 </instruction>
 <instruction order="29" opcode="SUB">
  <arg1 type="var">GF@x</arg1>
  <arg2 type="float">50.6</arg2>
  <arg3 type="float">10.4</arg3>
 </instruction>
 <instruction order="30" opcode="WRITE">
  <arg1 type="var">GF@x</arg1>
 </instruction>
 <instruction order="31" opcode="WRITE">
  <arg1 type="string">||</arg1>
 </instruction>
 <instruction order="32" opcode="MUL">
  <arg1 type="var">GF@x</arg1>
  <arg2 type="var">GF@a</arg2>
  <arg3 type="var">GF@b</arg3>
 </instruction>
 <instruction order="33" opcode="WRITE">
  <arg1 type="var">GF@x</arg1>
 </instruction>
 <instruction order="34" opcode="WRITE">
  <arg1 type="string">|</arg1>
 </instruction>
 <instruction order="35" opcode="MUL">
  <arg1 type="var">GF@x</arg1>
  <arg2 type="float">50.4</arg2>
  <arg3 type="var">GF@b</arg3>
 </instruction>
 <instruction order="36" opcode="WRITE">
  <arg1 type="var">GF@x</arg1>
 </instruction>
 <instruction order="37" opcode="WRITE">
  <arg1 type="string">|</arg1>
 </instruction>
 <instruction order="38" opcode="MUL">
  <arg1 type="var">GF@x</arg1>
  <arg2 type="var">GF@b</arg2>
  <arg3 type="float">50.5</arg3>
 </instruction>
 <instruction order="39" opcode="WRITE">
  <arg1 type="var">GF@x</arg1>
 </instruction>
 <instruction order="40" opcode="WRITE">
  <arg1 type="string">|</arg1>
 </instruction>
 <instruction order="41" opcode="MUL">
  <arg1 type="var">GF@x</arg1>
  <arg2 type="float">-2.4</arg2>
  <arg3 type="float">50.6</arg3>
 </instruction>
 <instruction order="42" opcode="WRITE">
  <arg1 type="var">GF@x</arg1>
 </instruction>
 <instruction order="43" opcode="WRITE">
  <arg1 type="string">||</arg1>
 </instruction>
 <instruction order="44" opcode="IDIV">
  <arg1 type="var">GF@x</arg1>
  <arg2 type="var">GF@a</arg2>
  <arg3 type="var">GF@b</arg3>
 </instruction>
 <instruction order="45" opcode="WRITE">
  <arg1 type="var">GF@x</arg1>
 </instruction>
 <instruction order="46" opcode="WRITE">
  <arg1 type="string">|</arg1>
 </instruction>
 <instruction order="47" opcode="IDIV">
  <arg1 type="var">GF@x</arg1>
  <arg2 type="var">GF@a</arg2>
  <arg3 type="float">10.0</arg3>
 </instruction>
 <instruction order="48" opcode="WRITE">
  <arg1 type="var">GF@x</arg1>
 </instruction>
 <instruction order="49" opcode="WRITE">
  <arg1 type="string">|</arg1>
 </instruction>
 <instruction order="50" opcode="IDIV">
  <arg1 type="var">GF@x</arg1>
  <arg2 type="float">10.4</arg2>
  <arg3 type="var">GF@a</arg3>
 </instruction>
 <instruction order="51" opcode="WRITE">
  <arg1 type="var">GF@x</arg1>
 </instruction>
 <instruction order="52" opcode="WRITE">
  <arg1 type="string">|</arg1>
 </instruction>
 <instruction order="53" opcode="IDIV">
  <arg1 type="var">GF@x</arg1>
  <arg2 type="float">-5</arg2>
  <arg3 type="float">-2.2</arg3>
 </instruction>
 <instruction order="54" opcode="WRITE">
  <arg1 type="var">GF@x</arg1>
 </instruction>
 <instruction order="55" opcode="WRITE">
  <arg1 type="string">||</arg1>
 </instruction>
</program>
