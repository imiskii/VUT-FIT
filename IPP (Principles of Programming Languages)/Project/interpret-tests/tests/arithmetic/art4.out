7.9|21.3|21.3|61.0||13.1|-20.7|0.3000000000000007|40.2||-27.3|-131.04|-131.3|-121.44||-4.038461538461538|1.05|0.9904761904761905|2.2727272727272725||